parachain/pallets/basic-channel/src/inbound/mod.rs

```diff
@@ -117,7 +118,7 @@ pub mod pallet {
 				}
 			})?;
 
-			let message_id = MessageId::new(ChannelId::Basic, envelope.nonce);
+			let message_id = MessageId::Basic(ChannelId::Basic, envelope.origin, envelope.nonce);
 			T::MessageDispatch::dispatch(envelope.source, message_id, &envelope.payload);
```

Comment:
@vgeddes We call the source address `origin` in the [dispatch call](https://github.com/Snowfork/snowbridge/blob/29a05937b5775be5b644f04888a75051be809484/parachain/pallets/dispatch/src/lib.rs#L121-L122). Is this a change in the meaning of "origin" in the dispatch pallet vs the outbound contract?

ie. in the contract "source" is an app and "origin" is a user, while the dispatch pallet uses "origin" to mean the app that sent the message.

Reply:
> We call the source address origin in the [dispatch call](https://github.com/Snowfork/snowbridge/blob/29a05937b5775be5b644f04888a75051be809484/parachain/pallets/dispatch/src/lib.rs#L121-L122).

Yeah that's purely a Substrate thing, where an [origin](https://docs.substrate.io/build/origins/) is something that invoked a pallet's dispatchable function. For example, a user could submit a transaction that ends up invoking a pallet dispatchable. Then that user's account is the origin.

In the code you've linked to, we are actually doing something very custom and not seen on many other Substrate chains. The dispatch pallet implements a custom origin which is an ethereum account ID (H160). When the dispatch pallet is asked to dispatch a message, it calls the dispatchable function (encoded in the message), and passes the the contract address of the Ethereum app as the origin.

This is important part of our security model. Our app pallets such as EthApp, DotApp, etc, need to ensure that their pallet dispatchables are being called only by their peer apps on the Ethereum side.

For example, in the code at https://github.com/Snowfork/snowbridge/blob/820ec5d8d8e1a9973790e05ccc9f7d3ca88fac53/parachain/pallets/eth-app/src/lib.rs#L151, the Ethapp::mint() dispatchable checks that the origin is a trusted peer app (EthApp.sol).